client/src/app/+videos/+video-watch/video-watch.component.ts

```diff
@@ -188,7 +188,14 @@ export class VideoWatchComponent implements OnInit, OnDestroy {
     })
 
     this.queryParamsSub = this.route.queryParams.subscribe(queryParams => {
-      this.playlistPosition = queryParams[ 'playlistPosition' ]
+      // Handle the ?playlistPosition
+      let position = queryParams[ 'playlistPosition' ]
+      if (position === 'last') position = -1 // Handle the "last" index
+      if (typeof position === 'string') position = parseInt(position, 10) // Sanitization needed for the "reverse" index
+      if (isNaN(position)) position = 1
+
+      if (this.playlist && position < 0) this.playlistPosition = this.playlist.videosLength - position + 1
```

Comment:
`this.playlist.videosLength + position + 1` since `position < 0` and try to use a ternary:

```
this.playlistPosition = this.playlist && position < 0
  ? ...
  : ...
```